session/session.go

```diff
@@ -553,7 +553,7 @@ func (s *session) ExecRestrictedSQL(sctx sessionctx.Context, sql string) ([]chun
 	)
 	// Execute all recordset, take out the first one as result.
 	for i, rs := range recordSets {
-		tmp, err := drainRecordSet(ctx, rs)
+		tmp, err := drainRecordSet(ctx, sctx, rs)
```

Comment:
Maybe it is better to use `se` here because it is the session that is used to execute the sql.